src/context/Build.zig

```diff
@@ -68,4 +70,27 @@ pub const Builtins = struct {
             return context.assetFind(ref, .{ .build = null });
         }
     };
+
+    pub const git = struct {
+        pub const signature: Signature = .{ .ret = .Git };
+        pub const description =
+            \\Returns git-related metadata if you are inside a git repository.
+            \\If you are not or the parsing failes, it will return an error.
+            \\Packed object are not supported, commit anything to get the metadata.
+        ;
+        pub const examples =
+            \\<div :text="$build.git"></div>
+        ;
+        pub fn call(
+            build: *const Build,
+            _: Allocator,
+            _: []const Value,
+        ) Value {
+            return build.git();
+        }
+    };
 };
+
+pub fn git(build: Build) Value {
```

Comment:
Unless I missed something and this function is called from somewhere else, please pull this logic directly into `call()`.

Reply:
Additionally, please implement a second scripty builtin called `git?`. The idea is for `git` to produce an error when the information is not present, and for `git?` to produce an optional value set to null when the information is missing.

This is done in order to give the use the choice of:

- assert that the value must be present and get an error otherwise
- only make use of the value if it is present (by using an `:if` attribute), and not error out in case its missing.


I will make a similar remark to branch and tag info from within the `git` value.

Reply:
To help you get right the optional version of the function, take a look at other pre-existing examples: it's important to always return a `context.Optional` in both the present and missing case.